src/game/Spell.cpp

```diff
@@ -2982,9 +2982,12 @@ void Spell::cast(bool skipCheck)
     // CAST SPELL
     SendSpellCooldown();
 
-    TakePower();
-    TakeReagents();                                         // we must remove reagents before HandleEffects to allow place crafted item in same slot
-    TakeAmmo();
+    if (!m_IsTriggeredSpell)
```

Comment:
Hey, it wasn't a diss in any way, but since you created this you may still have some idea how it fits together and how reliant the other changes are on that wrongful check there, or how valid the other changes are at all.